### ISSUE TITLE
Inventory gets messed up with DungeonsXL and Multiverse.

### ISSUE THREAD

**Issue by ZadronZSX** [elBukkit/MagicPlugin#820]
Whenever a player with their spell inventory/hotbar open enters an instanced DungeonsXL dungeon or gets teleported to another world at all, their inventory gets replaced with spell icon items. For now I had to put signs everywhere on the server telling everyone not to go to other worlds or go through any portals while their spell hotbar/inventory is open. Doing so also does swaps multiverse inventories sometimes. Sorry for the low res video, my graphics card died a few months ago so I am forced to record using integrated graphics.

[https://youtu.be/5tH1wmTJ2es](url)

**Comment by NathanWolf** [comment]
There is a fix for the world changing, try `/mconfig configure config close_wand_on_world_change: true`

However for any sort of plugin issue there will not be a fix unless I integrate with DungeonsXL or they integrate with Magic. Unless the issue is strictly the world change and Multiverse and nothing specific to going into a dungeon.

If you need to, you can set wands to chest mode with `/mconfig configure wand wand mode chest`

**Comment by ZadronZSX** [comment]
That doesn't fix it. Please look into integration with Multiverse Inventories and DungeonsXL whenever you have time. Thank you Nathan!

**Comment by NathanWolf** [comment]
Did that not even fix the issue with changing worlds? If so something may be wrong. That option was specifically meant for dealing with Multiverse and other per-world-inventory plugins.

**Comment by ZadronZSX** [comment]
Maybe it's an issue with the Multiverse Inventories plugin then.

**Comment by TheDarthFather** [comment]
Same happends to me. In this case is with a TNT plugin.
Not sure if this helps but ...
What I noticed is that it happens only if you have the wand 'opened'. 
If you have the wand on the task bar, but you have sometjing else selected it works fine.

**Comment by NathanWolf** [comment]
Strange, does the TNT plugin do anything weird with the inventory?

Magic always has problems with plugins that wand to modify your inventory, put a GUI directly in it, save it and clear it, that kind of thing. I can't imagine what a TNT plugin would be doing like that though?

MVInventories is not a huge surprise given what it does, but the option to close wands when switching worlds is meant to fix that. I have no thad a chance to try it out myself.

**Comment by TheDarthFather** [comment]
Yes, you are right -> both TNT and BedWars plugins I have are chaging some hotbar items with GUI ones.
And when I am joining any of them with the 'open' wand this inventory "confusion" happens.

**Comment by NathanWolf** [comment]
Ah that makes sense then :(
I can look and see if there’s some way I can integrate, but unless they fire an event whenever they’re about to swap out the inventory it’s going to be problematic.

An alternative is to use the spellbook, or switch wands to chest modes- there are ways to cast spells that don’t involve storing the the survival inventory.

I’d love to re-work Magic at some point to use pure packets for the spell inventory, but there are various problems with that approach.
